crawl-ref/source/stairs.cc

```diff
@@ -627,23 +635,13 @@ void floor_transition(dungeon_feature_type how,
         how = branches[old_level.branch].entry_stairs;
     }
 
-    // Special messages on returning from portal vaults, Abyss, Pan, etc.
-    if (!is_connected_branch(old_level.branch)
-        && player_in_connected_branch()
-        && old_level.branch != you.where_are_you)
-    {
-        mprf("Welcome %sto %s!",
-             you.chapter == CHAPTER_POCKET_ABYSS ? "" : "back ",
-             branches[you.where_are_you].longname);
-    }
-
-    // Falling down the stairs or portal.
+    // Check for falling down the stairs or portal. (Why can't you fall in the abyss?)
```

Comment:
Oh yeah, I found this curious. Anyone know why we check for abyss here? The only thing I could think of was that maybe it would be weird (and cruel) to make the player trip down the stairs if they were banished. But if that's the concern, I think it'd be better to just check `forced`.